src/WebJobs.Script.Grpc/Channel/GrpcWorkerChannel.cs

```diff
@@ -736,29 +735,9 @@ internal async Task SendInvocationRequest(ScriptInvocationContext context)
                     context.CancellationToken.Register(() => SendInvocationCancel(invocationRequest.InvocationId));
                 }
 
-                if (!string.IsNullOrEmpty(_workerCapabilities.GetCapabilityState(RpcWorkerConstants.EnableHttpProxying)))
+                if (!string.IsNullOrEmpty(_workerCapabilities.GetCapabilityState(RpcWorkerConstants.EnableHttpProxying)) && context.FunctionMetadata.IsHttpTriggerFunction())
```

Comment:
Does this change depend on any change on the language worker side? Or is it safe to deploy this host changes when we are ready?

Reply:
The changes on the host side and the code paths we are adding shouldn't be hit unless a worker has the capability set, but I think we're still far from deploying any of these host changes. These are just for the feature branch.